aya/src/programs/mod.rs

```diff
@@ -670,6 +682,46 @@ impl_try_from_program!(
     FExit,
 );
 
+macro_rules! impl_as_extension {
+    ($($ty:ident),+ $(,)?) => {
+        $(
+            impl AsRef<Extension> for $ty {
+                fn as_ref(&self) -> &Extension {
+                    // Safety: These types both encapsulate the same underlying data
+                    unsafe { & *(self as * const $ty as * const Extension) }
```

Comment:
😱

I had missed this! I don't think we need it. Unless I'm missing something `Extension` can store `program: &'a Program` and use `program.data()` to access `ProgramData`

Reply:
Removed it and adjusted the API per discord discussion